core-bundle/src/Security/Voter/AbstractFrontendAccessVoter.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Voter;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\FrontendUser;
+use Contao\Model;
+use Contao\StringUtil;
+use Contao\System;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+abstract class AbstractFrontendAccessVoter extends Voter implements ServiceSubscriberInterface
+{
+    public const ATTRIBUTE = 'contao_frontend.access';
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @internal
+     * @required
+     */
+    public function setContainer(ContainerInterface $container): ?ContainerInterface
+    {
+        $previous = $this->container;
+        $this->container = $container;
+
+        return $previous;
+    }
+
+    public static function getSubscribedServices()
+    {
+        return [
+            'contao.framework' => ContaoFramework::class,
+        ];
+    }
```

Comment:
What about using the `ServiceSubscriberTrait` instead and providing a `contaoFramework() : ContaoFramework` method?